src/figuresClasses.ts

```diff
@@ -1,19 +1,94 @@
-export interface Figure {
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: string;
+  color: Color;
+  radius?: number;
+  sides?: number[];
+  getArea: () => number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
 
-}
+  public sides: number[] = [];
+
+  constructor(public color: Color, ...sides: number[]) {
```

Comment:
let's move the logic with throwing errors to a separate method - and call it inside the constructor